### ISSUE TITLE
EditContext: how is bold represented?

### ISSUE THREAD

**Issue by annevk** [w3c/editing#368]
The `textformatupdate` event doesn't seem able to represent a variety of inputs. Am I missing something?

**Comment by annevk** [comment]
It was pointed out to me this might be based on a misunderstanding of the existing styles, which are there for IMEs. Still, I'm curious to know how styling of text would be mediated through `EditContext`.

**Comment by alexkeng** [comment]
EditContext only handles plain texts (it provides plain text view to communicate with OS text input services), so the styling of text like bold, italic, etc, are out of scope of EC, and should be handled by the author's editor model.

The textformatupdate event here is not for styling of the text, it's for the styling of the "decoration" that IME provides to give users hints during the composition, ex showing phrases in Japanese IME (the active phrase has thick underline)
![image](https://user-images.githubusercontent.com/42681140/145543258-b9a0105f-303c-4ce7-b8b4-13d04b8b8163.png)

with this information, the author can render the decoration (ex. in a canvas element) to re-create the same IME experience as in normal text input box.

The current spec includes the styles that Windows TSF supports (the [TF_DISPLAYATTRIBUTE ](https://docs.microsoft.com/en-us/windows/win32/api/msctf/ns-msctf-tf_displayattribute) structure). The spec doesn't support all the styles defined in [NSUnderlineStyle](https://developer.apple.com/documentation/uikit/nsunderlinestyle?language=objc) on Mac (ex. alternating dashes and dots, NSUnderlineStylePatternDashDot), but those special styles don't seem to be used by IMEs, so the current set of styles should be good.

**Comment by johanneswilm** [comment]
When used with contenteditable, it is my understanding that could combine it with the Input Events for formatting changes in order to make use of those native browser menus @annevk mentions. Isn't that right @alexkeng?

**Comment by alexkeng** [comment]
@johanneswilm, EditContext is not expected to be used with contenteditable, if we attach an EditContext to a contenteditable div, EditContext's behavior will override contenteditable, i.e. the user input won't change DOM, instead it will go to EditContext's text buffer and fire related events.

@annevk, EditContext is designed to be able to integrate with native editing UI like the iOS menu you mentioned. We can think of EditContext as a special version of contenteditable, all the editing heuristics remain the same, e.g. the editing UI, the context menu, etc, the only difference is that the user input now doesn't go to the div and instead it goes to the EditContext. And only after the author's editor model handles the EditContext events and updates the DOM do the users see the characters they type. So back to your question, the native editing UI doesn't need to be reimplemented, but after user apply some command in the menu, e.g. bold, a 'formatBold' input event will be fired, and the authors are responsible to update the DOM accordingly.

**Comment by annevk** [comment]
I see, so where is it defined that `formatBold` can be dispatched and what interface it will use, etc.?

**Comment by johanneswilm** [comment]
@alexkeng right now, the way the input events spec is written, the kind of **editing host** that is being used influences a bit how the events are used. The spec only mentions contenteditable, input type "text" and textarea [1]. Now if this is to work like an editing host without being one of those three, then we probably need to mention that somewhere. Is it to behave exactly like contenteditable when it comes to (before)input events? So that means that some OSes will show the bold, underline, etc. menus when it obtains focus? Is there any way to turn those off or individual menu items off? How about the options provided with contenteditable-disabled? Is that to be integrated in some way? 

I see that setting `spellcheck=false` or `spellcheck=true` on the element will work the way they do with contenteditable? How about adding noneditable islands inside of that element using contenteditable=false? How about an island with a lake like the one below?

```
<div contenteditable="true">
<p>blablabla...</p>
<figure contenteditable="false">
<img src="...">
<figcaption contenteditable="true">The caption of the figure</figcaption>
</figure>
<p>blablabla...</p>
</div>
```

Until now it was my understanding that all these extra formatting menus, input events, etc. would only appear if an element is both an EditContext AND a contenteditable element at the same time. If that is not the case, then I wonder exactly how that would work.


[1] https://w3c.github.io/input-events/#overview
